core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/DedupAggregator.java

```diff
@@ -144,21 +146,23 @@ private Schema getOutputSchema(Schema inputSchema) {
   }
 
   private void validateSchema(Schema inputSchema, List<String> uniqueFields,
-                              @Nullable DedupConfig.DedupFunctionInfo function) {
+                              @Nullable DedupConfig.DedupFunctionInfo function, FailureCollector collector) {
     for (String uniqueField : uniqueFields) {
       Schema.Field field = inputSchema.getField(uniqueField);
       if (field == null) {
-        throw new IllegalArgumentException(String.format("Cannot do an unique on field '%s' because it does " +
-                                                           "not exist in inputSchema '%s'", uniqueField, inputSchema));
+        collector.addFailure(String.format("Cannot do an unique on field '%s' because it does " +
```

Comment:
This message is overly verbose, let's change it to:

Field '%s' does not exist in the input schema

Reply:
Done